src/bp-members/bp-members-functions.php

```diff
@@ -309,16 +309,8 @@ function bp_core_get_userid_from_nickname( $nickname ) {
 		return false;
 	}
 
-	$users = get_users(
-		array(
-			'meta_key'    => 'nickname',
-			'meta_value'  => $nickname,
-			'number'      => 1,
-			'count_total' => false,
-		)
-	);
-
-	$user = $users ? $users[0] : null;
+	// Get member from subsite on multisite env.
+	$user = get_user_by( 'slug', $nickname );
```

Comment:
PROD-928 #comment @kishanjasani, @dipskakadiya   This is not a valid change.
`slug` will check the column from the table `user_nicename` 
originally we fetch from user meta with key `nickname` which is different than `user_nicename`